doc/ampi/manual.rst

```diff
@@ -213,79 +213,198 @@ Non-blocking collectives have been defined in AMPI since before
-be a thread on which one of more AMPI ranks are scheduled. We define a
-process here as an operating system process, which may contain one or
-more worker threads. The built-in attributes are ``AMPI_MY_WTH``,
-``AMPI_MY_PROCESS``, ``AMPI_NUM_WTHS``, and ``AMPI_NUM_PROCESSES``.
-These attributes are accessible from any rank by calling
-``MPI_Comm_get_attr``, such as:
+The build script for Charm++ is called ``build``. The syntax for this
+script is:
 
-.. code-block:: fortran
+.. code-block:: bash
 
-   ! Fortran:
-   integer :: my_wth, flag, ierr
-   call MPI_Comm_get_attr(MPI_COMM_WORLD, AMPI_MY_WTH, my_wth, flag, ierr)
+   $ build <target> <version> <opts>
 
+For building AMPI (which also includes building Charm++ and other
+libraries needed by AMPI), specify ``<target>`` to be ``AMPI``. And
+``<opts>`` are command line options passed to the ``charmc`` compile
+script. Common compile time options such as
+``-g, -O, -Ipath, -Lpath, -llib`` are accepted.
 
-.. code-block:: c++
+To build a debugging version of AMPI, use the option: ``-g``. To build a
+production version of AMPI, use the option: ``--with-production``.
 
-   // C/C++:
-   int my_wth, flag;
-   MPI_Comm_get_attr(MPI_COMM_WORLD, AMPI_MY_WTH, &my_wth, &flag);
+``<version>`` depends on the machine, operating system, and the
+underlying communication library one wants to use for running AMPI
+programs. See the charm/README file for details on picking the proper
+version. Here is an example of how to build a debug version of AMPI in a
+linux and ethernet environment:
 
-AMPI also provides extra communicator types that users can pass to
-``MPI_Comm_split_type``: ``AMPI_COMM_TYPE_HOST`` for splitting a
-communicator into disjoint sets of ranks that share the same physical
-host, ``AMPI_COMM_TYPE_PROCESS`` for splitting a communicator into
-disjoint sets of ranks that share the same operating system process, and
-``AMPI_COMM_TYPE_WTH``, for splitting a communicator into disjoint sets
-of ranks that share the same worker thread.
+.. code-block:: bash
 
-For parsing Fortran command line arguments, AMPI Fortran programs should
-use our extension APIs, which are similar to Fortran 2003’s standard
-APIs. For example:
+   $ ./build AMPI netlrts-linux-x86_64 -g
 
-.. code-block:: fortran
+And the following is an example of how to build a production version of
+AMPI on a Cray XC system, with MPI-level error checking in AMPI turned
+off:
 
-   integer :: i, argc, ierr
-   integer, parameter :: arg_len = 128
-   character(len=arg_len), dimension(:), allocatable :: raw_arguments
+.. code-block:: bash
 
-   call AMPI_Command_argument_count(argc)
-   allocate(raw_arguments(argc))
-   do i = 1, size(raw_arguments)
-       call AMPI_Get_command_argument(i, raw_arguments(i), arg_len, ierr)
-   end do
+   $ ./build AMPI gni-crayxc --with-production --disable-ampi-error-checking
 
-Name for Main Program
+AMPI can also be built with support for shared memory on any
```

Comment:
could mention here what shared memory is used for

Reply:
Done